feature/experimental/gribi/otg_tests/programming_with_reload/programming_with_reload_test.go

```diff
@@ -0,0 +1,534 @@
+	innerIPHeader := flow.Packet().Add().Ipv4()
+	innerIPHeader.Src().SetValue(innerSrcIP1)
+	innerIPHeader.Dst().Increment().SetStart(innerDstIP1).SetCount(1)
+	flow.EgressPacket().Add().Ethernet()
+	return flow
+}
+
+// validateTrafficFlows verifies that the flow on ATE, traffic should pass for good flow and fail for bad flow.
+func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, good []gosnappi.Flow, bad []gosnappi.Flow) {
+	top := ate.OTG().FetchConfig(t)
+	top.Flows().Clear()
+	for _, flow := range append(good, bad...) {
+		top.Flows().Append(flow)
+	}
+	ate.OTG().PushConfig(t, top)
+
+	ate.OTG().StartProtocols(t)
+	ate.OTG().StartTraffic(t)
+
+	time.Sleep(15 * time.Second)
+	ate.OTG().StopTraffic(t)
+	time.Sleep(10 * time.Second)
+	otgutils.LogFlowMetrics(t, ate.OTG(), top)
+
+	for _, flow := range good {
+		outPkts := float32(gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().OutPkts().State()))
+		inPkts := float32(gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().InPkts().State()))
+		if outPkts == 0 {
+			t.Fatalf("OutPkts for flow %s is 0, want > 0", flow)
+		}
+		if got := ((outPkts - inPkts) * 100) / outPkts; got > 0 {
+			t.Fatalf("LossPct for flow %s: got %v, want 0", flow.Name(), got)
+		}
+	}
+
+	for _, flow := range bad {
+		outPkts := float32(gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().OutPkts().State()))
+		inPkts := float32(gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().InPkts().State()))
+		if outPkts == 0 {
+			t.Fatalf("OutPkts for flow %s is 0, want > 0", flow)
+		}
+		if got := ((outPkts - inPkts) * 100) / outPkts; got < 100 {
+			t.Fatalf("LossPct for flow %s: got %v, want 100", flow.Name(), got)
+		}
+	}
+}
+
+// setDUTInterfaceState sets the admin state on the dut interface
+func (args *testArgs) setDUTInterfaceWithState(t testing.TB, p *ondatra.Port, state bool) {
+	dc := gnmi.OC()
+	i := &oc.Interface{}
+	i.Enabled = ygot.Bool(state)
+	i.Type = ethernetCsmacd
+	i.Name = ygot.String(p.Name())
+	gnmi.Update(t, args.dut, dc.Interface(p.Name()).Config(), i)
+}
+
+// addStaticRoute configures static route needed for decap path
+func addStaticRoute(t *testing.T, dut *ondatra.DUTDevice, ip string) {
+	d := gnmi.OC()
+	s := &oc.Root{}
+	static := s.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut)).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut))
+	ipv4Nh := static.GetOrCreateStatic(innerDstIP1 + "/" + mask).GetOrCreateNextHop("0")
+	ipv4Nh.NextHop, _ = ipv4Nh.To_NetworkInstance_Protocol_Static_NextHop_NextHop_Union(ip)
+	gnmi.Update(t, dut, d.NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut)).Config(), static)
+}
+
+// Reload chassis
+func (args *testArgs) reloadChassis(t *testing.T) {
+
+	preRebootCompStatus := gnmi.GetAll(t, args.dut, gnmi.OC().ComponentAny().OperStatus().State())
```

Comment:
Any chance we can include the actual components here instead of simply printing their oper-status and looking if the number of components is the same. I  raised something for gNOI-3.1 in https://github.com/openconfig/featureprofiles/pull/2397